gnu-getopt 2.32.1 (update and change source)

okdana opened the pull request:
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

The current `gnu-getopt` formula is derived from the project hosted by the original developer, which is seemingly no longer maintained — it has been merged into util-linux, and needed bug fixes can be found only there. Homebrew has a `util-linux` formula, but it can't provide `getopt` because it (`getopt`) has to be keg-only. So what i've done is rework the `gnu-getopt` formula to *also* build from util-linux.

I couldn't find anything in the Formula Cookbook about multiple formulae derived from the same project, nor any existing examples in the repo, so i'm not sure if there's a policy for this. My main concern is that updates to util-linux now have to be applied in two different places, which is kind of weird for maintenance. But they're completely independent of each other, so i guess in practice it'll probably work the same as any other formula — when someone notices that an update to `gnu-getopt` is needed, they'll make the change.

Also, `util-linux`'s bottle section doesn't include Yosemite or Mavericks, and i assumed there was a technical reason for that, so i dropped them here too.

Is this OK?

okdana commented:
Actually, `curl` and `curl-openssl` are kind of the same idea as this (two formulae from the same project), so i guess there's precedent after all.

zbeekman commented on Formula/gnu-getopt.rb:
```diff
@@ -1,31 +1,34 @@
+# developer, which has since been merged into util-linux. The util-linux formula
+# could not be repurposed to provide getopt, though, since it only includes
+# tools that aren't already shipped with macOS. Thus, we now have two formulae
+# that offer independent subsets of util-linux
```
This comment text should be removed from the formula.

You can add it in the commit message below the standard formatting

zbeekman commented on Formula/gnu-getopt.rb:
```diff
@@ -1,31 +1,34 @@
     sha256 "05391b0dd0876ead74b18a4c5bb9c7db996586bf6918bd014db534027fd9ae2a" => :sierra
     sha256 "5e9e87fe18c5681e80f1cf940fed275ed895831304326bc5e7be6fb6e53e8594" => :el_capitan
-    sha256 "f8dbbec03aaaeb1bc774d9bf606701901cc9a8ad15cecc5473567e51845057e6" => :yosemite
-    sha256 "27938c615808c8e4ff2eacac0a4059c76dee5518a5c8bbfb304b24b70736b429" => :mavericks
```
Please don't edit the `bottle do` block. This gets updated manually during the CI build process. If it needs adjustment after that it is the responsibility of maintainers to take care of that.

zbeekman commented on Formula/gnu-getopt.rb:
```diff
@@ -1,31 +1,34 @@
 
   keg_only :provided_by_macos
 
-  depends_on "gettext"
```
Are you certain `gettext` is not required? Have you tested building and installing locally after uninstalling `gettext`?

okdana commented:
Thanks. Fixed the things you mentioned and rebased.

As far as `gettext`, i removed that because the `util-linux` formula doesn't use it. I think you *can* build it against `gettext`, but it isn't required. This is what it looks like when i build from the new formula without it:

```
% brew uninstall --ignore-dependencies gettext
Uninstalling /usr/local/Cellar/gettext/0.19.8.1... (1,935 files, 16.9MB)
% brew uninstall --force gnu-getopt
Uninstalling gnu-getopt... (10 files, 153.8KB)
% brew install --build-from-source gnu-getopt
==> Downloading https://www.kernel.org/pub/linux/utils/util-linux/v2.32/util-linux-2.32.1.tar.xz
Already downloaded: /Users/dana/Library/Caches/Homebrew/downloads/6b0e8abc3964fd06808baaff231cda1875526dd2a86b757eb142f816f0bb1685--util-linux-2.32.1.tar.xz
==> ./configure --disable-silent-rules --prefix=/usr/local/Cellar/gnu-getopt/2.32.1
==> make getopt
==> Caveats
gnu-getopt is keg-only, which means it was not symlinked into /usr/local,
because macOS already provides this software and installing another version in
parallel can cause all kinds of trouble.

If you need to have gnu-getopt first in your PATH run:
  echo 'export PATH="/usr/local/opt/gnu-getopt/bin:$PATH"' >> ~/.zshrc


Bash completion has been installed to:
  /usr/local/opt/gnu-getopt/etc/bash_completion.d
==> Summary
🍺  /usr/local/Cellar/gnu-getopt/2.32.1: 10 files, 153.7KB, built in 1 minute 8 seconds
% /usr/local/opt/gnu-getopt/bin/getopt --version
getopt from util-linux 2.32.1
```

zbeekman commented:
hmmm this looks like a problem with our CI infra. I don't have time to fix the CI issue, gut this is good to go, as far as I am concerned... We'll see if any other @Homebrew/maintainers have issues with this, but I vote to ship it.

sjackman commented:
```
Error Message

failed: brew doctor
Stacktrace

        Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: A newer Command Line Tools release is available.
Update them from Software Update in System Preferences.
```
https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/40445/version=mojave/testReport/brew-test-bot/mojave/doctor/

CI only failed on Mojave with this `brew doctor` failure. I believe we can safely ignore this failure and merge this PR.

zbeekman commented:
@okdana: yes it’s a CI issue on our end. I’ll send this through for testing again once it’s fixed.

zbeekman commented:
@BrewTestBot test this please

zbeekman commented:
@BrewTestBot test this please

zbeekman commented:
@BrewTestBot test this please

zbeekman commented:
Thanks for your contribution to Homebrew, @okdana! 🎉 🥇 

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!
